flink-contrib/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -1991,43 +1999,71 @@ public int numStateEntries() {
 		return count;
 	}
 
+	/**
+	 * This class is not thread safety.
+	 */
 	private static class RocksIteratorWrapper<K> implements Iterator<K> {
```

Comment:
One change that I would suggest to something that was not introduced in the PR: we could make `RocksIteratorWrapper` implement `AutoCloseable` and use that close method instead of calling close directly on the RockDB iterator where this is used. I think this is cleaner, because the wrapper should own the Rocks iterator.

Reply:
Good point, addressed.